state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -470,6 +457,8 @@ pub mod pallet {
 
 		/// Safe Mode access.
 		type SafeMode: Get<PalletSafeMode<I>>;
+
+		type SwapLimitsProvider: SwapRequestValidationProvider;
```

Comment:
As dicussed, we should keep the names consistent. Also, `SwapRequestValidationProvider` isn't very clear, I think I would prefer `SwapLimitsProvider` even if, as you say, it doesn't provide *all* the limits.